File: sling-react-core/src/main/java/com/cathcart93/sling/models/ImageGalleryItem.java
package com.cathcart93.sling.models;

import com.google.gson.annotations.Expose;
import org.apache.sling.api.resource.Resource;
import org.apache.sling.models.annotations.Model;
import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;

@Model(adaptables = Resource.class)
public class ImageGalleryItem {
    @Expose
    @ValueMapValue(name = "original", optional = true)
    private String original;

    @Expose
    @ValueMapValue(name = "thumbnail", optional = true)
    private String thumbnail;
}

File: sling-react-components/src/main/kotlin/com/cathcart93/sling/components/models/ImageGalleryItem.kt
package com.cathcart93.sling.components.models

import com.cathcart93.sling.core.ReactProp
import org.apache.sling.api.resource.Resource
import org.apache.sling.models.annotations.Model
import org.apache.sling.models.annotations.injectorspecific.ValueMapValue

@Model(adaptables = arrayOf(Resource::class))
class ImageGalleryItem {
    @ValueMapValue(name = "original", optional = true)
    @ReactProp
    private val original: String? = null

    @ValueMapValue(name = "thumbnail", optional = true)
    @ReactProp
    private val thumbnail: String? = null
}
